modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/store/FilePageStoreManager.java

```diff
@@ -312,25 +298,22 @@ private Path ensureGroupWorkDir(String grpName) throws IgniteInternalCheckedExce
         return groupWorkDir;
     }
 
-    private GroupPageStoreHolder<FilePageStore> createGroupFilePageStoreHolder(
+    private List<FilePageStore> createGroupFilePageStoreHolder(
             String grpName,
             int partitions
     ) throws IgniteInternalCheckedException {
         Path groupWorkDir = ensureGroupWorkDir(grpName);
 
         FilePageStoreFactory filePageStoreFactory = new FilePageStoreFactory(filePageStoreFileIoFactory, pageSize);
 
-        FilePageStore idxFilePageStore = filePageStoreFactory.createPageStore(TYPE_IDX, groupWorkDir.resolve(INDEX_FILE_NAME));
-
         FilePageStore[] partitionFilePageStores = new FilePageStore[partitions];
```

Comment:
Why do you need an array here if you copy it into a List? Can we simply use an `ArrayList` here?

Reply:
Fix it